scrapy/utils/sitemap.py

```diff
@@ -34,10 +37,13 @@ def __iter__(self):
                 yield d
 
 
-def sitemap_urls_from_robots(robots_text):
+def sitemap_urls_from_robots(robots_text, base_url=None):
     """Return an iterator over all sitemap urls contained in the given
     robots.txt file
     """
     for line in robots_text.splitlines():
         if line.lstrip().lower().startswith('sitemap:'):
-            yield line.split(':', 1)[1].strip()
+            url = line.split(':', 1)[1].strip()
+            if not re.match('https?://', url, re.I):
```

Comment:
As a matter of fact it does, but I see (at least) one case in which the output would not be _exactly_ the same: if we remove that check and just `yield urljoin(base_url, url)`, if the sitemap has a line like `Sitemap: HTTP://example.com/sitemap.xml`, the yielded URL would be `http://example.com/sitemap.xml` (lowercase).

I guess it's a matter of preference, @redapple, @kmike what would you guys like to do?